internal/wire/parse.go

```diff
@@ -939,8 +939,8 @@ func (pt ProvidedType) IsNil() bool {
 	return pt.p == nil && pt.v == nil && pt.a == nil
 }
 
-// ConcreteType returns the concrete type that was provided.
-func (pt ProvidedType) ConcreteType() types.Type {
+// Type returns the concrete type that was provided.
```

Comment:
I'm not sure that I understand the rationale behind this change. While it's true that a provider might return an interface type, this will never be a `wire.Bind`-ed type, for instance. `ConcreteType` still conveys the intended meaning to me, but regardless, it seems this doc comment needs to reflect more of what we want this to convey.

Reply:
Talked offline, name change is ok, updated the docstring.